test/src/unittests/tonal/test_audio2pitch.py

```diff
@@ -0,0 +1,227 @@
+        sample_rate: int,
+        amplitude_in_db: float,
+        frequency: float,
+        pitch_precision: float = 1,
+        conf_precision: float = 0.1,
+        loudness_precision: float = 0.1,
+    ):
+        frameSize = 1024
+        hopsize = frameSize
+
+        frames = FrameGenerator(signal, frameSize=frameSize, hopSize=hopsize)
+        pitchDetect = Audio2Pitch(frameSize=frameSize, sampleRate=sample_rate)
+        n_outputs = len(pitchDetect.outputNames())
+        pitch, confidence, loudness, voiced = ([] for _ in range(n_outputs))
+        for frame in frames:
+            f, conf, l, v = pitchDetect(frame)
+            pitch += [f]
+            confidence += [conf]
+            loudness += [amp2db(l)]
+            voiced += [v]
+        self.assertAlmostEqual(mean(f), frequency, pitch_precision)
+        self.assertAlmostEqual(mean(confidence), 1, conf_precision)
+        self.assertAlmostEqual(mean(loudness), amplitude_in_db, loudness_precision)
+        self.assertAlmostEqual(mean(voiced), 1, conf_precision)
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(Audio2Pitch(), {"frameSize": 1})
+        self.assertConfigureFails(Audio2Pitch(), {"sampleRate": 0})
+        self.assertConfigureFails(
+            Audio2Pitch(), {"sampleRate": 44100, "maxFrequency": 44100}
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "maxFrequency": 200, "minFrequency": 250},
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "pitchAlgorithm": "yin_fft"},
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "loudnessAlgorithm": "ebur128"},
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "loudnessThreshold": 1.0},
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "pitchConfidenceThreshold": -0.5},
+        )
+        self.assertConfigureFails(
+            Audio2Pitch(),
+            {"sampleRate": 44100, "pitchConfidenceThreshold": 1.5},
+        )
+
+    def testARealCase(self):
+        # The expected values were recomputed from commit
+        # 2d37c0713fb6cc5f637b3d8f5d65aa90b36d4277
+        #
+        # The expeted values were compared with the vamp pYIN
+        # implementation of the YIN algorithm producing very
+        # similar values.
+        #
+        # https://code.soundsoftware.ac.uk/projects/pyin
+
+        frameSize = 1024
+        sample_rate = 44100
+        hopSize = 512
+        loudness_threshold = -80
+        filename = join(testdata.audio_dir, "recorded", "vignesh.wav")
+        if sys.platform == "darwin":
+            import soundfile as sf
+
+            audio, _ = sf.read(filename, dtype="float32")
+        else:
+            audio = MonoLoader(filename=filename, sampleRate=44100)()
```

Comment:
What is the rationale for using soundfile (libsndfile) instead of MonoLoader specifically on MacOS?

Reply:
because at least in my machine MonoLoader and all algos using FFmpeg library are not working. So, it is the only way I had to run and pass all the tests.

Reply:
Is the FFmpeg issue solved in others OS versions?

Reply:
Ideally, we should resolve the FFmpeg issue, however it is out of scope of this PR. We can remove this custom workaround (soundfile) and merge this PR, plus create a new GitHub issue related to the problem you found.

Reply:
okay! let's do that.